pppp.c

```diff
@@ -497,16 +534,44 @@ int startup()
 	printf("HTTP connection from %s:%d\n",
 		inet_ntoa(iclientip->sin_addr), ntohs(iclientip->sin_port));
 	while (1) {
-		len = read(client, inbuf, sizeof(inbuf));
-		if (len < 7) {
-			close(client);
-			goto again;
-		}
-		if (strncasecmp(inbuf, "GET ", 4)) {
-			close(client);
-			goto again;
-		}
-		if (!strncmp(inbuf+4, "/v.mjpg ", 8)) {
+        char inbuf[1024];
+        int len = read(client, inbuf, sizeof(inbuf));
+        if (len <= 7) {
+            close(client);
+            goto again;
+        }
+		// if (strncasecmp(inbuf, "GET ", 4)) {
+		// 	close(client);
+		// 	goto again;
+		// }
+        if (!strncmp(inbuf+4, "/exec ", 7)) {
+            char *json_end = strstr(inbuf+7, "\r\n");
+            if (json_end) {
+                *json_end = '\0'; // Null terminate the JSON string
+                sendExecCmd(inbuf+7);
+            } else {
+                fprintf(stderr, "Invalid JSON format\n");
+            }
+    		// Send HTTP status line
+			char status_line[256];
+			sprintf(status_line, "HTTP/1.1 200 OK\r\n");
+			send(client_fd, status_line, strlen(status_line), 0);
+
+			// Send content type header
+			char content_type[256];
+			sprintf(content_type, "Content-Type: application/json\r\n");
+			send(client_fd, content_type, strlen(content_type), 0);
+
+			// Send blank line to indicate end of headers
+			char blank_line[] = "\r\n";
+			send(client_fd, blank_line, strlen(blank_line), 0);
```

Comment:
calling strlen() for string constants is just stupid. The length is already known at compile time.